libraries/resource_provider.rb

```
@@ -137,7 +137,7 @@ def remove
     def patch
       invalid_params = @context.operation.nil? || @context.path.nil?
       raise "InvalidParameters: Parameters 'operation' and 'path' must be set for patch" if invalid_params
-      @item.retrieve!
+      return Chef::Log.error("Skipping patch since #{@resource_name} '#{@name}' does not exists") unless @item.retrieve!
```

@tmiotto , won't we want it to just fail hard here if it can't be retrieved? I'm imagining it will run into a strange "uri not set" error on the patch if not, right?

---

Actually it will return at this point.

Alas, I really don't know what option would suit best in this situation.

---

Idempotently talking, it does not make sense to skip the update if resource was not found. Actually, if I apply a patch to an existant resource, its state should be the resource with the patch applied, if it doesn't exist, I cannot create it to make hte resulting state the same.